docs/src/advanced.md

```diff
@@ -0,0 +1,233 @@
+```
+
+## Filtering Test Files
+
+You can also use [`find_tests`](@ref) to automatically discover test files and then filter or modify them.
+This requires manually parsing arguments so that filtering is only applied when the user did not request specific tests to run:
+
+```julia
+# Start with autodiscovered tests
+testsuite = find_tests(pwd())
+
+# Parse arguments
+args = parse_args(ARGS)
+
+if filter_tests!(testsuite, args)
+    # Remove tests that shouldn't run on Windows
+    if Sys.iswindows()
+        delete!(testsuite, "ext/specialfunctions")
+    end
+end
+
+runtests(MyModule, args; testsuite)
+```
+
+The [`filter_tests!`](@ref) function returns `true` if no positional arguments were provided (allowing additional filtering) and `false` if the user specified specific tests (preventing further filtering).
+
+## Initialization Code
+
+Use the `init_code` keyword argument to [`runtests`](@ref) to provide code that runs before each test file.
+This is useful for:
+- Importing packages
+- Defining constants, defaults or helper functions
+- Setting up test infrastructure
+
+```julia
+using ParallelTestRunner
+
+const init_code = quote
+    using Test
+    using MyPackage
+
+    # Define a helper function available to all tests
+    function test_helper(x)
+        return x * 2
+    end
+end
+
+runtests(MyPackage, ARGS; init_code)
+```
+
+The `init_code` is evaluated in each test's sandbox module, so all definitions are available to your test files.
+
+## Custom Workers
+
+For tests that require specific environment variables or Julia flags, you can use the `test_worker` keyword argument to assign tests to custom workers:
+
+```julia
+using ParallelTestRunner
+
+function test_worker(name)
+    if name == "needs_env_var"
+        # Create a worker with a specific environment variable
+        return addworker(; env = ["SPECIAL_ENV_VAR" => "42"])
+    elseif name == "needs_threads"
+        # Create a worker with multiple threads
+        return addworker(; exeflags = ["--threads=4"])
+    end
+    # Return nothing to use the default worker
+    return nothing
+end
+
+testsuite = Dict(
+    "needs_env_var" => quote
+        @test ENV["SPECIAL_ENV_VAR"] == "42"
+    end,
+    "needs_threads" => quote
+        @test Base.Threads.nthreads() == 4
+    end,
+    "normal_test" => quote
+        @test 1 + 1 == 2
+    end
+)
+
+runtests(MyPackage, ARGS; test_worker, testsuite)
+```
+
+The `test_worker` function receives the test name and should return either:
+- A worker object (from [`addworker`](@ref) for tests that need special configuration
+- `nothing` to use the default worker pool
+
+## Custom Output Streams
+
+You can redirect output to custom I/O streams:
+
+```julia
+using ParallelTestRunner
+
+io = IOBuffer()
+runtests(MyPackage, ARGS; stdout=io, stderr=io)
+
+# Process the output
+output = String(take!(io))
+```
+
+This is useful for:
+- Capturing test output for analysis
+- Writing to log files
+- Suppressing output in certain contexts
```

Comment:
This is technically not false, but I don't think it's really useful for end-users, it's mainly for testing the features of this package itself.  I'm going to remove also this section.